File: backend/internal/domain/idea.go
package domain

type Idea struct {
	ID               string  `db:"id" json:"id"`
	VentureID        string  `db:"venture_id" json:"venture_id"`
	RawInput         string  `db:"raw_input" json:"raw_input"`
	OneLineConcept   *string `db:"one_line_concept" json:"one_line_concept,omitempty"`
	TargetCustomer   *string `db:"target_customer" json:"target_customer,omitempty"`
	ValueProposition *string `db:"value_proposition" json:"value_proposition,omitempty"`
	KeyAssumptions   *string `db:"key_assumptions" json:"key_assumptions,omitempty"`
	EarlyRisks       *string `db:"early_risks" json:"early_risks,omitempty"`
	Version          int     `db:"version" json:"version"`
	IsLocked         bool    `db:"is_locked" json:"is_locked"`
	Status           string  `db:"status" json:"status"`
	AiRawInput       *string `db:"ai_raw_input" json:"-"`
	AiRawOutput      *string `db:"ai_raw_output" json:"-"`
	CreatedAt        string  `db:"created_at" json:"created_at"`
	UpdatedAt        string  `db:"updated_at" json:"updated_at"`
}

type StructuredConcept struct {
	OneLineConcept   string   `json:"one_line_concept"`
	TargetCustomer   string   `json:"target_customer"`
	ValueProposition string   `json:"value_proposition"`
	KeyAssumptions   []string `json:"key_assumptions"`
	EarlyRisks       []string `json:"early_risks"`
}

type UpdateIdeaRequest struct {
	OneLineConcept   *string `json:"one_line_concept,omitempty"`
	TargetCustomer   *string `json:"target_customer,omitempty"`
	ValueProposition *string `json:"value_proposition,omitempty"`
	KeyAssumptions   *string `json:"key_assumptions,omitempty"`
	EarlyRisks       *string `json:"early_risks,omitempty"`
}

File: frontend/lib/domain/models/idea.dart
class Idea {
  final String id;
  final String ventureId;
  final String rawInput;
  final String? oneLineConcept;
  final String? targetCustomer;
  final String? valueProposition;
  final String? keyAssumptions;
  final String? earlyRisks;
  final int version;
  final bool isLocked;
  final String status;
  final String createdAt;
  final String updatedAt;

  Idea({
    required this.id,
    required this.ventureId,
    required this.rawInput,
    this.oneLineConcept,
    this.targetCustomer,
    this.valueProposition,
    this.keyAssumptions,
    this.earlyRisks,
    required this.version,
    required this.isLocked,
    required this.status,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Idea.fromJson(Map<String, dynamic> json) => Idea(
        id: json['id'] as String,
        ventureId: json['venture_id'] as String,
        rawInput: json['raw_input'] as String,
        oneLineConcept: json['one_line_concept'] as String?,
        targetCustomer: json['target_customer'] as String?,
        valueProposition: json['value_proposition'] as String?,
        keyAssumptions: json['key_assumptions'] as String?,
        earlyRisks: json['early_risks'] as String?,
        version: json['version'] as int,
        isLocked: json['is_locked'] as bool,
        status: json['status'] as String,
        createdAt: json['created_at'] as String,
        updatedAt: json['updated_at'] as String,
      );
}
